src/client/game.cpp

```diff
@@ -3887,7 +3888,15 @@ void Game::updateFrame(ProfilerGraph *graph, RunStats *stats, f32 dtime,
 		runData.update_draw_list_timer = 0;
 		client->getEnv().getClientMap().updateDrawList();
 		runData.update_draw_list_last_cam_dir = camera_direction;
+		draw_list_updated = true;
+	}
+
+	if (ShadowRenderer *shadow = RenderingEngine::get_shadow_renderer())
+		update_draw_list_delta = shadow->getUpdateDelta();
 
+	if (runData.update_shadows_timer > update_draw_list_delta &&
+			(!draw_list_updated || RenderingEngine::get_shadow_renderer()->getDirectionalLightCount() == 0)) {
```

Comment:
Could this be reached even when there is no shadow renderer?

Reply:
I disabled shadow update entirely when shadow renderer is not there in 603a70d.